micromanager_gui/_core_widgets/_presets_widget.py

```diff
@@ -0,0 +1,139 @@
+
+    def _on_combo_changed(self, text: str) -> None:
+        self._mmc.setConfig(self._group, text)
+        set_wdg_color(self.text_color, self._combo)
+
+    def _on_cfg_set(self, group: str, preset: str) -> None:
+        if group == self._group and self._combo.currentText() != preset:
+            with signals_blocked(self._combo):
+                self._combo.setCurrentText(preset)
+                set_wdg_color(self.text_color, self._combo)
+
+    def value(self) -> str:
+        return self._combo.currentText()
+
+    def setValue(self, value: str) -> None:
+        if value not in self._mmc.getAvailableConfigs(self._group):
+            raise ValueError(
+                f"{value!r} must be one of {self._mmc.getAvailableConfigs(self._group)}"
+            )
+        self._combo.setCurrentText(str(value))
+
+    def allowedValues(self) -> Tuple[str]:
+        return tuple(self._combo.itemText(i) for i in range(self._combo.count()))
+
+    def refresh(self) -> None:
+        with signals_blocked(self._combo):
+            self._combo.clear()
+            if self._group not in self._mmc.getAvailableConfigGroups():
+                self._combo.addItem(f"No group named {self._group}.")
+                self._combo.setEnabled(False)
+            else:
+                presets = self._mmc.getAvailableConfigs(self._group)
+                self._combo.addItems(presets)
+                self._combo.setEnabled(True)
+
+    def _highlight_if_preset_state_changed(self, group: str, preset: str):
+        """Set the text color to magenta if a preset has changed"""
+
+        if group == self._group:
+            return
+
+        dp = get_dev_prop(group, preset)
+        if any(x for x in dp if x in self.dev_prop):
+            self._set_if_props_match_preset()
+
+    def _highlight_if_prop_changed(self, device: str, property: str, value: str):
+        """Set the text color to magenta if a preset property has changed"""
+
+        if (device, property) not in self.dev_prop:
+            if self._mmc.getDeviceType(device) != DeviceType.StateDevice:
+                return
+            # a StateDevice has also a "Label" property. If "Label" is not
+            # in dev_prop, we check if the property "State" is in dev_prop.
+            if (device, "State") not in self.dev_prop:
+                return
+
+        self._set_if_props_match_preset()
+
+    def _set_if_props_match_preset(self):
+        for p in self._presets:
+            _set_combo = True
+            dpv = get_dev_prop_val(self._group, p)
+            for i in dpv:
+                cache_prop = self._mmc.getPropertyFromCache(i[0], i[1])
+                if cache_prop != i[2]:
+                    _set_combo = False
+                    break
+            if _set_combo:
+                with signals_blocked(self._combo):
+                    self._combo.setCurrentText(p)
+                    set_wdg_color(self.text_color, self._combo)
+                    return
```

Comment:
`get_dev_prop_val` is only used here and is just one line so we can inline it and also use some more explicit variable names (to help reviewers like me ;) )

```suggestion
        for preset in self._presets:
            _set_combo = True
            for (dev, prop, config_value) in self._mmc.getConfigData(self._group, p):
                cache_value = self._mmc.getPropertyFromCache(dev, prop)
                if cache_value != config_value:
                    _set_combo = False
                    break
            if _set_combo:
                with signals_blocked(self._combo):
                    self._combo.setCurrentText(preset)
                    set_wdg_color(self.text_color, self._combo)
                    return
        # None of the presets match the current system state
        set_wdg_color("magenta", self._combo)
```


You can tighten up this logic a bit. Also using the redoing of the loop in the comment above it's nice to use more explicit names that `i[0], i[1]`